CONTRIBUTION.md

```diff
@@ -0,0 +1,47 @@
+# Contribution guidlines
+## Preamble
+These guidelines are intended as "guidelines" to how an PR To Dire's Buildinggadgets should be structures. 
+Bear in mind that these are only guidelines and don't necessarily apply in each and every case, but if you choose not to follow a specific Part of this specification, you must be able to reason out why, if asked to do so.
+## PR structure
+### General Notes
+In General a PR should explain what it does, how it does this and how it was tested. It should also follow the [Code Guidelines](#code-guidelines).
+* If it is a Bug fix, prepend it with \[Bug]/\[Fix#xyz].
+* If you are not yet finished implementing, prepend it with \[WIP]
+### Bug Fix PR's
+Your PR should include the following at least the following Information:
+* A reference to the corresponding issue or a full description of the issue being fixed
+* An explanation how this PR solves/attempts to solve the given issue
+* If testing was performed, what tests and how extensive they have been
+### Feature Request PR's
+Your PR should include the following at least the following Information:
+* A reference to the corresponding issue or a full description of the feature being implemented
+* An explanation how this PR solves/attempts to solve the given issue
+* If testing was performed, what tests and how extensive they have been
+## Code Guidelines
+In General your code should be well-structured and easy to read. 
+We know that as of 1.12 this is not the case for great parts of the mod, but we are trying to improve upon the code-base.
+Also your PR should not contain any formatting only changes to Files, as this simply makes reviewing them unnecessarily complex.
+### Formatting
+* Use 4 space indent instead of Tabs
+* `{` should not be prefix with returns
+* Methods should have exactly one free line between them (/their docs)
+* Avoid empty lines at the start of class, interface or enum definitions
+* operators should be surrounded by spaces
+* a `,` comma to separate types and type casts should be followed by exactly one Space
+* Method/Class annotations should gain their own lines in front of the definition
+* Parameter Annotations should be surrounded by exactly one space
+* Avoid unnecessary lines in empty Methods
+    * `public void emptyImpl(){ }`
```

Comment:
```suggestion
    * `public void emptyImpl() {}`
```
There should always be a space between a method's `)` and `{`. And the lack of space between `{` and `}` is more in-line with the code base.